tests/filecheck/backend/riscv/memref_to_riscv_opt.mlir

```diff
@@ -0,0 +1,24 @@
+// RUN: xdsl-opt -p convert-memref-to-riscv,reconcile-unrealized-casts,canonicalize %s  | filecheck %s
+
+// Test that a memref store and load with constant indices optimise to a single operation
+
+builtin.module {
+    %v_reg, %m_reg = "test.op"() : () -> (!riscv.freg<>, !riscv.reg<>)
+    %v = builtin.unrealized_conversion_cast %v_reg : !riscv.freg<> to f32
+    %m = builtin.unrealized_conversion_cast %m_reg : !riscv.reg<> to memref<3x2xf32>
+    %r_reg = riscv.li 1 : () -> !riscv.reg<>
+    %c_reg = riscv.li 1 : () -> !riscv.reg<>
+    %r = builtin.unrealized_conversion_cast %r_reg : !riscv.reg<> to index
+    %c = builtin.unrealized_conversion_cast %c_reg : !riscv.reg<> to index
+    "memref.store"(%v, %m, %r, %c) {"nontemporal" = false} : (f32, memref<3x2xf32>, index, index) -> ()
+    %x = "memref.load"(%m, %r, %c) {"nontemporal" = false} : (memref<3x2xf32>, index, index) -> (f32)
```

Comment:
Does this also work for ints?

Reply:
The lowering from memref also handles integer types at least (https://github.com/xdslproject/xdsl/blob/8f06da92e84347a59764cc4fbc462fa6c1a37b95/docs/Toy/toy/rewrites/lower_memref_riscv.py#L108).
It should be also quite straightforward to add the optimization for `lw/sw'.

Reply:
I added a comment for this on #1507

Reply:
Definitely related to this PR, but I'd rather add it in a follow-up if that's ok with you?